woocommerce/woocommerce
# Remove Woo AI Enabled Setting and rework copy on the Woo AI Settings page.

### Issue
We created a Woo AI Settings page which has a toggle enabling the "Woo AI Features", however, this toggle didn't have any effect on logic within the plugin; it simply was showing/hiding settings on the settings page.

### Changes proposed in this Pull Request:

- Rewrite some of the Woo AI Settings page copy so that it is more clear that it's for enabling Personalization of the AI generated content. It doesn't totally disable AI features (you should just disable the entire plugin for that).

![Screenshot 2023-08-17 at 12 26 35](https://github.com/woocommerce/woocommerce/assets/6723003/b34d17a9-3deb-44b0-9b23-7eb6e4b7cef9)

- Only generate product descriptions using this information if it's filled out (tone of voice not neutral).
- Rewrite the getAllBrandingSettings() hook to get all settings with a single network request rather than multiple.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce->Settings->Advanced->Features and enable AI Personalization.
2. Change the tone of voice to something not neutral and add a business description.
3. Go to Products->Add New and generate a description. You should see that this data is added to the prompt:

![Screen Shot on 2023-08-17 at 12:42:56](https://github.com/woocommerce/woocommerce/assets/6723003/9c989f0c-052e-494d-ab9e-1c8c5320fcc4)
You should also see that the output is affected by having personalization applied.
4. Go back to WooCommerce->Settings->Advanced->Features and clear out the settings (set tone of voice to neutral)
5. Go to Products->Add New and generate a description. You should see that this data is _not_ added to the prompt, resulting in a more neutral output.
![Screenshot 2023-08-17 at 12 36 08](https://github.com/woocommerce/woocommerce/assets/6723003/7233e398-44fe-47a8-bbc7-ce451041c004)

### Changelog entry




<details>

#### Significance




#### Type




#### Message Add Woo AI Personalization enabled setting and check setting when generating descriptions with AI.

</details>
